clients/slo/slo.go

```diff
@@ -0,0 +1,212 @@
+	}
+	defer resp.Body.Close()
+
+	return api.ProcessResponse(resp)
+}
+
+func (c *Client) Create(ctx context.Context, body json.RawMessage) (api.Response, error) {
+	resp, err := c.restClient.POST(ctx, endpointPath, bytes.NewReader(body), rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsg, "create", err)
+	}
+	defer resp.Body.Close()
+
+	return api.ProcessResponse(resp)
+}
+
+func (c *Client) Update(ctx context.Context, id string, body json.RawMessage) (api.Response, error) {
+	if id == "" {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, errors.New("argument \"id\" is empty"))
+	}
+
+	getResp, err := c.Get(ctx, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	version, err := getOptimisticLockingVersion(getResp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	path, err := url.JoinPath(endpointPath, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	resp, err := c.restClient.PUT(ctx, path, bytes.NewReader(body), rest.RequestOptions{
+		CustomShouldRetryFunc: rest.RetryIfTooManyRequests,
+		QueryParams:           url.Values{"optimistic-locking-version": []string{version}},
+	})
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+	defer resp.Body.Close()
+
+	return api.ProcessResponse(resp)
+}
+
+func (c *Client) Delete(ctx context.Context, id string) (api.Response, error) {
+	if id == "" {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, errors.New("argument \"id\" is empty"))
+	}
+
+	getResp, err := c.Get(ctx, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	version, err := getOptimisticLockingVersion(getResp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	path, err := url.JoinPath(endpointPath, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	resp, err := c.restClient.DELETE(ctx, path, rest.RequestOptions{
+		CustomShouldRetryFunc: rest.RetryIfTooManyRequests,
+		QueryParams:           url.Values{"optimistic-locking-version": []string{version}},
+	})
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+	defer resp.Body.Close()
+
+	return api.ProcessResponse(resp)
+}
+
+const errMsg = "failed to %s slo resource: %w"
+const errMsgWithId = "failed to %s slo resource with id %s: %w"
+
+func getOptimisticLockingVersion(resp api.Response) (string, error) {
+	var body struct {
+		Version string `json:"version"`
+	}
+
+	err := json.Unmarshal(resp.Data, &body)
+	if err != nil {
+		return "", err
+	}
+
+	return body.Version, nil
+}
+
+func processListResponse(httpResponse *http.Response) (string, api.ListResponse, error) {
```

Comment:
For 3 or more return values, please add names. I have no idea what the first argument here is without looking at the code

Reply:
done